File: NativeAPI/Python/b2_start_large_file.py
import requests
import json

api_url = ""  # Provided by b2_authorize_account
account_authorization_token = ""  # Provided by b2_authorize_account
bucket_id = ""  # Provided by b2_list_buckets
file_name = ""  # The name of the file
content_type = ""  # Content Type of the file

response = requests.post(
    f"{api_url}/b2api/v4/b2_start_large_file",
    headers={"Authorization": account_authorization_token},
    json={
        "bucketId": bucket_id,
        "fileName": file_name,
        "contentType": content_type
    }
)
response_data = response.json()
print(json.dumps(response_data, indent=2))

File: NativeAPI/Java/B2StartLargeFile.java
package com.backblaze.examples;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class B2StartLargeFile {
    public static void main(String[] args) {
        String apiUrl = ""; // Provided by b2_authorize_account
        String accountAuthorizationToken = ""; // Provided by b2_authorize_account
        String bucketId = ""; // The ID of the destination bucket. The bucket ID is provided by b2_create_bucket, b2_list_buckets.
        String fileName = ""; // Name of the file as it will appear in B2
        String contentType = ""; // Content type of the file

        String postParams = "{" +
                "\"bucketId\":\"" + bucketId + "\"," +
                "\"fileName\":\"" + fileName + "\"," +
                "\"contentType\":\"" + contentType + "\"" +
                "}";

        try (HttpClient client = HttpClient.newHttpClient()) {
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create(apiUrl + "/b2api/v4/b2_start_large_file"))
                    .header("Authorization", accountAuthorizationToken)
                    .POST(HttpRequest.BodyPublishers.ofString(postParams))
                    .build();
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            String jsonResponse = response.body();
            System.out.println(jsonResponse);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
